Modules/StudyProgramme/classes/class.ilObjStudyProgrammeMembersGUI.php

```diff
@@ -244,52 +282,260 @@ protected function _addUsers($a_users) {
 		return $assignments;
 	}
 
-	public function markAccredited() {
+	/**
+	 * Mark SP for single user accredited
+	 *
+	 * @return null
+	 */
+	public function markAccredited()
+	{
 		require_once("Modules/StudyProgramme/classes/class.ilStudyProgrammeUserProgress.php");
-		$prgrs = $this->getProgressObject();
-		$prgrs->markAccredited($this->user->getId());
+		$prgrs_id = $this->getPrgrsId();
+		$this->accredit($prgrs_id);
 		$this->showSuccessMessage("mark_accredited_success");
 		$this->ctrl->redirect($this, "view");
 	}
-	
-	public function unmarkAccredited() {
+
+	/**
+	 * Mark SP for users accredited
+	 *
+	 * @return null
+	 */
+	public function markAccreditedMulti()
+	{
+		$prgrs_ids = $_POST["prgs_ids"];
+		foreach ($prgrs_ids as $key => $prgrs_id) {
+			$this->accredit((int)$prgrs_id);
+		}
+		$this->showSuccessMessage("mark_accredited_multi_success");
+		$this->ctrl->redirect($this, "view");
+	}
+
+	/**
+	 * Accredited SP
+	 *
+	 * @param int 	$prgrs_id
+	 *
+	 * @return null
+	 */
+	protected function accredit($prgrs_id)
```

Comment:
I think I understand why there is a `markAccredited` and an `accredit` but this duplication seems odd when reading the code. What would be the difference between accrediting something and marking something as accredited? Please rename `accredit` to `markAccreditedById` to clarify that there indeed is no difference. Same thing for `deaccredit` and `remove` please.